src/Microsoft.AspNet.Mvc.TagHelpers/InputTagHelper.cs

```diff
@@ -236,25 +236,27 @@ private void GenerateCheckBox(ModelExplorer modelExplorer, TagHelperOutput outpu
                 }
             }
 
-            var tagBuilder = Generator.GenerateCheckBox(
+            var checkBoxTag = Generator.GenerateCheckBox(
                 ViewContext,
                 modelExplorer,
                 For.Name,
                 isChecked: null,
                 htmlAttributes: htmlAttributes);
-            if (tagBuilder != null)
+            if (checkBoxTag != null)
             {
                 // Do not generate current element's attributes or tags. Instead put both <input type="checkbox"/> and
                 // <input type="hidden"/> into the output's Content.
                 output.Attributes.Clear();
                 output.TagName = null;
 
-                output.Content.Append(tagBuilder.ToHtmlContent(TagRenderMode.SelfClosing));
+                checkBoxTag.TagRenderMode = TagRenderMode.SelfClosing;
```

Comment:
Sorry, I'm not understanding this "Won't fix". The HTML specification makes it clear that end tags are illegal for void elements e.g. `<input>`. The HTML generator deals with such HTML semantics. Why instead spread the information around?

Separately I mentioned leaving handling of start and end tags alone in my original comment. That's a separate concern from whether the end tag is valid at all.